server/broadcast.go

```diff
@@ -584,6 +584,7 @@ func transcodeSegment(cxn *rtmpConnection, seg *stream.HLSSegment, name string,
 				now := time.Now()
 				uri, err := drivers.SaveRetried(bros, name, data, map[string]string{"duration": segDurMs}, 2)
 				took := time.Since(now)
+				recordWG.Done()
```

Comment:
Is FlushRecord() supposed to be called after all the segments have been inserted into the JSON playlist? If so, should this Done() call be after the InsertHLSSegmentJSON() call below?

Reply:
Good catch, fixed.